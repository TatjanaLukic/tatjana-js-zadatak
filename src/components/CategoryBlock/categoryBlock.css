@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;0,700;1,400&display=swap");
.category-block {
  margin: 1rem 0.5rem;
}
@media screen and (min-width: 576px) {
  .category-block {
    margin: 1rem 0.5rem;
  }
}

.category-block__link {
  display: inline-block;
  text-transform: capitalize;
}
@media screen and (min-width: 576px) {
  .category-block__link {
    margin-left: 4.5rem;
    margin-top: 0.5rem;
    font-size: 1.3rem;
  }
}

.category-block__slides--wrapper {
  display: flex;
  flex-direction: column;
  background-color: #343a40;
  margin-top: 0.5rem;
  padding: 0.5rem 0;
}
@media screen and (min-width: 576px) {
  .category-block__slides--wrapper {
    flex-direction: row;
  }
}

.category-block__slides {
  margin: auto;
  width: 100%;
  background-color: white;
  padding: 0 0.3rem;
  min-height: 260px;
}
@media screen and (min-width: 576px) {
  .category-block__slides {
    min-height: 280px;
  }
}
@media screen and (min-width: 992px) {
  .category-block__slides {
    margin: auto;
    width: 90%;
    min-height: 360px;
  }
}
@media screen and (min-width: 1200px) {
  .category-block__slides {
    margin: auto;
    width: 75%;
    min-height: 300px;
  }
}

.category-block__nav {
  font-size: bold;
  font-size: 1.8rem;
  width: 10%;
  margin: auto;
  text-align: center;
  color: white;
  cursor: pointer;
}
@media screen and (min-width: 576px) {
  .category-block__nav {
    width: 7rem;
    font-size: bold;
    font-size: 4rem;
  }
}

.category-block--hide {
  display: none;
}

.category-block--show {
  display: block;
}

.category-block__icon--show {
  display: inline-block;
  margin: 0.5rem;
  cursor: pointer;
}

.category-block__icon--hide {
  display: none;
}

/*# sourceMappingURL=categoryBlock.css.map */
