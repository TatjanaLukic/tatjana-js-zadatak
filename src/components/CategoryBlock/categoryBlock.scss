@import "../../assets/variables";
.category-block {
  margin: 1rem 0.5rem;

  @media screen and (min-width: $smallSize) {
    margin: 1rem 0.5rem;
  }
}

.category-block__link {
  display: inline-block;
  text-transform: capitalize;

  @media screen and (min-width: $smallSize) {
    margin-left: 4.5rem;
    margin-top: 0.5rem;
    font-size: 1.3rem;
  }
}

.category-block__slides--wrapper {
  display: flex;
  flex-direction: column;
  background-color: $slider-background-color;
  margin-top: 0.5rem;
  padding: 0.5rem 0;

  @media screen and (min-width: $smallSize) {
    flex-direction: row;
  }
}

.category-block__slides {
  margin: auto;
  width: 100%;
  background-color: $slide-background-color;
  padding: 0 0.3rem;
  min-height: 260px;

  @media screen and (min-width: $smallSize) {
    min-height: 280px;
  }

  @media screen and (min-width: $largeSize) {
    margin: auto;
    width: 90%;
    min-height: 360px;
  }

  @media screen and (min-width: $extraSize) {
    margin: auto;
    width: 75%;
    min-height: 300px;
  }
}

.category-block__nav {
  font-size: bold;
  font-size: 1.8rem;
  width: 10%;
  margin: auto;
  text-align: center;
  color: $slider-nav-color;
  cursor: pointer;

  @media screen and (min-width: $smallSize) {
    width: 7rem;
    font-size: bold;
    font-size: 4rem;
  }
}

.cb__nav--disable {
  opacity: 0.3;
}

.category-block--hide {
  display: none;
}
.category-block--show {
  display: block;
}

.category-block__icon--show {
  display: inline-block;
  margin: 0.5rem;
  cursor: pointer;
}

.category-block__icon--hide {
  display: none;
}
