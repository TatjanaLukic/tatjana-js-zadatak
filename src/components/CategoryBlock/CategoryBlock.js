import React, { useContext, useEffect, useState } from "react";
import axiosInstance from "../../utils/axiosInstance";
import CountryContext from "../../context/countryContext";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import { getUrl } from "../../utils/helpers";
import Slide from "../Slide/Slide";
import { ReactComponent as IconExpand } from "../../assets/images/expand.svg";
import { ReactComponent as IconCollapse } from "../../assets/images/collapse.svg";
import Loading from "../../components/Loading/Loading";

import "./categoryBlock.css";

const CategoryBlock = ({ categoryName }) => {
  const country = useContext(CountryContext);

  const [active, setActive] = useState(true);
  const [currentSlide, setCurrentSlide] = useState(0);
  const [news, setNews] = useState([]);
  const [loading, setLoading] = useState(true);
  const maxSlides = 5;

  useEffect(() => {
    const fetchCategoryNews = async () => {
      try {
        const response = await axiosInstance.get(
          getUrl(country.country, categoryName, maxSlides)
        );

        const data = response.data;
        setNews(data.articles);
        setLoading(false);
      } catch (error) {
        console.log(error);
      }
    };

    fetchCategoryNews();
  }, [country.country, categoryName]);

  const onClickRight = () => {
    if (currentSlide < 4) {
      setCurrentSlide(currentSlide + 1);
    }
  };

  const onClickLeft = () => {
    if (currentSlide > 0) {
      setCurrentSlide(currentSlide - 1);
    }
  };

  const toggleCategory = () => {
    setActive(!active);
  };

  const newsSlides = news.map((data, inx) => {
    return (
      <Slide
        key={inx}
        isCurrent={inx === currentSlide}
        title={data.title}
        urlToImage={data.urlToImage}
        description={data.description}
        author={data.author}
        content={data.content}
        publishedAt={data.publishedAt}
        parentPath="/CategoriesPage"
      />
    );
  });

  if (loading) {
    return <Loading />;
  }

  return (
    <div className="category-block" data-testid="categoryBlock">
      <Link
        className="category-block__link"
        to={{
          pathname: `/CategoriesPage/TopNewsPage`,
          state: { categoryName },
        }}
      >
        {categoryName}
      </Link>

      <span
        className={
          active ? "category-block__icon--show" : "category-block__icon--hide"
        }
        onClick={toggleCategory}
        data-testid="collapse"
      >
        <IconCollapse />
      </span>

      <span
        className={
          active ? "category-block__icon--hide" : "category-block__icon--show"
        }
        onClick={toggleCategory}
        data-testid="expand"
      >
        <IconExpand />
      </span>

      <div className={active ? "category-block--show" : "category-block--hide"} data-testid="slider">
        <div className="category-block__slides--wrapper">
          <div
            className={
              currentSlide === 0
                ? "category-block__nav cb__nav--disable"
                : "category-block__nav"
            }
            data-testid="nav-left"
            onClick={onClickLeft}
          >
            &lt;
          </div>
          <div className="category-block__slides">{newsSlides}</div>
          <div
            className={
              currentSlide === maxSlides - 1
                ? "category-block__nav cb__nav--disable"
                : "category-block__nav"
            }
            data-testid="nav-right"
            onClick={onClickRight}
          >
            &gt;
          </div>
        </div>
      </div>
    </div>
  );
};

CategoryBlock.propTypes = {
  categoryName: PropTypes.string.isRequired,
};

export default CategoryBlock;
