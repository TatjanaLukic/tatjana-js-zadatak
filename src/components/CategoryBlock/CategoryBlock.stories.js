import React from "react";
import { MemoryRouter } from "react-router-dom";
import CategoryBlock from "./CategoryBlock";

export default {
  title: "App/CategoryBlock",
  component: CategoryBlock,
  decorators: [(getStory) => <MemoryRouter>{getStory()}</MemoryRouter>],
};

const Template = (args) => <CategoryBlock {...args} />;

export const Default = Template.bind({});
Default.args = {
    categoryName:"entertainment"
};
