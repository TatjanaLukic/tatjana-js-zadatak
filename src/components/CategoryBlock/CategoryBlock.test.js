import React from "react";
import { shallow } from "enzyme";
import CategoryBlock from "./CategoryBlock";

describe("<CategoryBlock />", () => {
  let categoryBlock;
  let data = {
    categoryName:"entertainment"
  };

  beforeEach(() => {
    categoryBlock = shallow(<CategoryBlock {...data} />);
  });

  it("should have 1 Link", () => {
    expect(categoryBlock.find("Link")).toHaveLength(1);
  });
  it("should have 5 Slide", () => {
    expect(categoryBlock.find("Slide")).toHaveLength(5);
  });
  it("spanshot matches", () => {
    expect(categoryBlock).toMatchSnapshot();
  });
});
