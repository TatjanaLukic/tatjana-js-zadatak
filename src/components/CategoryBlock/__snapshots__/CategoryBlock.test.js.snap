// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryBlock /> to match spanshots  1`] = `
<div>
  <div
    class="loading"
    data-testid="loading"
  >
    <svg>
      circles.svg
    </svg>
  </div>
</div>
`;

exports[`<CategoryBlock /> to match spanshots  2`] = `
<div>
  <div
    class="category-block"
    data-testid="categoryBlock"
  >
    <a
      class="category-block__link"
      href="/CategoriesPage/TopNewsPage"
    >
      business
    </a>
    <span
      class="category-block__icon--show"
      data-testid="collapse"
    >
      <svg>
        collapse.svg
      </svg>
    </span>
    <span
      class="category-block__icon--hide"
      data-testid="expand"
    >
      <svg>
        expand.svg
      </svg>
    </span>
    <div
      class="category-block--show"
      data-testid="slider"
    >
      <div
        class="category-block__slides--wrapper"
      >
        <div
          class="category-block__nav cb__nav--disable"
          data-testid="nav-left"
        >
          &lt;
        </div>
        <div
          class="category-block__slides"
        >
          <div
            class="slide-show"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Tesla Aktie schmiert ab   Kein Saft am Batterie Tag   BILD
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://bilder.bild.de/fotos/tesla aktie schmiert ab kein saft am batterie tag 201524804 73032638/Bild/3.bild.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Tesla Chef Elon Musk hat vor einer Veranstaltung zur Batterie Technologie die Erwartungen gedämpft und so die Aktienhändler enttäuscht.
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Strom selber machen? :Die fünf größten Irrtümer bei Solaranlagen   n tv NACHRICHTEN
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://bilder3.n tv.de/img/incoming/crop14785186/9961323629 cImg_16_9 w1200/imago61527841h.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Eigenheimbesitzer kommen am Thema Fotovoltaik kaum noch vorbei   schließlich soll eine Solaranlage nicht nur umweltfreundlich, sondern auch lukrativ sein. Wer sich über Solaranlagen informiert, stößt jedoch schnell auf widersprüchliche Aussagen. Ein Faktenche…
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Warum Airbus' Wasserstoffflieger hinten keine Fenster haben   aeroTELEGRAPH
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://cdn.aerotelegraph.com/production/uploads/2020/09/airbus_zero_e.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Zwei der drei Wasserstoffflieger, die Airbus präsentiert, unterscheiden sich auf den ersten Blick nicht sehr von heutigen Flugzeugen. Ein wichtiges Detail ist jedoch anders.
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Höhle der Löwen: Kölner Gründer klärt auf, Investoren geschockt   EXPRESS
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://www.express.de/image/37378240/2x1/940/470/a89bb94d2ab24631545ad42855d9b08f/NT/forest gum thomas kraemer.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              „Die Höhle der Löwen“ Investoren interessieren sich sehr für das Kölner Start up „Forest Gum“. Aber reicht es am Montag für einen Deal?
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Erste Fahrt im VW ID.4: Als Volkswagen einmal schneller als Tesla war   DER SPIEGEL
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://cdn.prod.www.spiegel.de/images/1c62a093 2f1a 41de a462 781e87c69868_w1280_r1.77_fpx33.34_fpy50.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Mit Tamtam stampft Tesla in Brandenburg die Fabrik fürs SUV Model Y aus dem Boden. Weniger beachtet: Bei VW rollen die ersten Exemplare des Konkurrenten ID.4 schon vom Band. Muss Elon Musk zittern? Eine Probefahrt.
            </p>
          </div>
        </div>
        <div
          class="category-block__nav"
          data-testid="nav-right"
        >
          &gt;
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<CategoryBlock /> to match spanshots  3`] = `
<div>
  <div
    class="category-block"
    data-testid="categoryBlock"
  >
    <a
      class="category-block__link"
      href="/CategoriesPage/TopNewsPage"
    >
      business
    </a>
    <span
      class="category-block__icon--hide"
      data-testid="collapse"
    >
      <svg>
        collapse.svg
      </svg>
    </span>
    <span
      class="category-block__icon--show"
      data-testid="expand"
    >
      <svg>
        expand.svg
      </svg>
    </span>
    <div
      class="category-block--hide"
      data-testid="slider"
    >
      <div
        class="category-block__slides--wrapper"
      >
        <div
          class="category-block__nav cb__nav--disable"
          data-testid="nav-left"
        >
          &lt;
        </div>
        <div
          class="category-block__slides"
        >
          <div
            class="slide-show"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Tesla Aktie schmiert ab   Kein Saft am Batterie Tag   BILD
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://bilder.bild.de/fotos/tesla aktie schmiert ab kein saft am batterie tag 201524804 73032638/Bild/3.bild.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Tesla Chef Elon Musk hat vor einer Veranstaltung zur Batterie Technologie die Erwartungen gedämpft und so die Aktienhändler enttäuscht.
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Strom selber machen? :Die fünf größten Irrtümer bei Solaranlagen   n tv NACHRICHTEN
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://bilder3.n tv.de/img/incoming/crop14785186/9961323629 cImg_16_9 w1200/imago61527841h.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Eigenheimbesitzer kommen am Thema Fotovoltaik kaum noch vorbei   schließlich soll eine Solaranlage nicht nur umweltfreundlich, sondern auch lukrativ sein. Wer sich über Solaranlagen informiert, stößt jedoch schnell auf widersprüchliche Aussagen. Ein Faktenche…
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Warum Airbus' Wasserstoffflieger hinten keine Fenster haben   aeroTELEGRAPH
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://cdn.aerotelegraph.com/production/uploads/2020/09/airbus_zero_e.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Zwei der drei Wasserstoffflieger, die Airbus präsentiert, unterscheiden sich auf den ersten Blick nicht sehr von heutigen Flugzeugen. Ein wichtiges Detail ist jedoch anders.
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Höhle der Löwen: Kölner Gründer klärt auf, Investoren geschockt   EXPRESS
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://www.express.de/image/37378240/2x1/940/470/a89bb94d2ab24631545ad42855d9b08f/NT/forest gum thomas kraemer.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              „Die Höhle der Löwen“ Investoren interessieren sich sehr für das Kölner Start up „Forest Gum“. Aber reicht es am Montag für einen Deal?
            </p>
          </div>
          <div
            class="slide-hide"
            data-testid="slide"
          >
            <h3
              class="slide__title"
            >
              <a
                class="slide__link"
                href="/CategoriesPage/NewsPage"
              >
                Erste Fahrt im VW ID.4: Als Volkswagen einmal schneller als Tesla war   DER SPIEGEL
              </a>
            </h3>
            <img
              alt="pictureOfNews"
              class="slide__image"
              src="https://cdn.prod.www.spiegel.de/images/1c62a093 2f1a 41de a462 781e87c69868_w1280_r1.77_fpx33.34_fpy50.jpg"
            />
            <p
              class="slide__desc"
              data-testid="description"
            >
              Mit Tamtam stampft Tesla in Brandenburg die Fabrik fürs SUV Model Y aus dem Boden. Weniger beachtet: Bei VW rollen die ersten Exemplare des Konkurrenten ID.4 schon vom Band. Muss Elon Musk zittern? Eine Probefahrt.
            </p>
          </div>
        </div>
        <div
          class="category-block__nav"
          data-testid="nav-right"
        >
          &gt;
        </div>
      </div>
    </div>
  </div>
</div>
`;
